config/ipfs_config.go

```diff
@@ -0,0 +1,34 @@
+package config
+
+// IPFSConfig defines a subset of the IPFS config that will be passed to the IPFS init and IPFS node (as a service)
+// spun up by the tendermint node.
+// It is mostly concerned about port configuration (Addresses).
+type IPFSConfig struct {
+	// is where the generated IPFS config and files will be stored.
+	// The default is ~/.tendermint/ipfs.
+	ConfigRootPath string
+	// TODO: can we avoid copying the fields from ipfs' config.Addresses here?
+	// TODO: also, these are only used on init. Maybe ConfigRootPath is sufficient?
+	API     string // address for the local API (RPC)
+	Gateway string // address to listen on for IPFS HTTP object gateway
+	// swarm related options:
+	Swarm      []string // addresses for the swarm to listen on
+	Announce   []string // swarm addresses to announce to the network
+	NoAnnounce []string // swarm addresses not to announce to the network
+}
```

Comment:
In this example from the doc (on how to use ipfs as a library), an initial set of bootstrap nodes to connect to is also provided:
https://github.com/ipfs/go-ipfs/blob/7588a6a52a789fa951e1c4916cee5c7a304912c2/docs/examples/go-ipfs-as-a-library/main.go#L294-L316

Depending on if we want to use a private network or not, we will have to do that as well. For now it seems the default config which we use (apart from the few options above) provides a good enough default set of bootstrap nodes.